lib/vrl/parser/src/lex.rs

```diff
@@ -447,7 +448,7 @@ impl StringLiteral<&str> {
 // -----------------------------------------------------------------------------
 
 impl<'input> Iterator for Lexer<'input> {
-    type Item = Spanned<'input, usize>;
+    type Item = SpannedResult<'input, usize>;
```

Comment:
Are these changes needed for the 1.50 upgrade, or did they too sneak in from a VRL merge?

Reply:
These ones are intentional. Clippy was complaining about `unnecessary_wraps` so I renamed the existing `Spanned` type `SpannedResult` and added a new `Spanned` type that represents the `Ok` value of `SpannedResult` so I could easily refer to `Spanned` in the function return types where it was complaining about the unnecessary wrapping. For example: https://github.com/timberio/vector/pull/6428/files/f279014f63a058abe29e0afb8dbc1f1dffc2fb32#diff-3eaef579e2d938c6926fa17614f80aab7455adeeffd8907567b34fee18a07e94R575-R582